### PR TITLE
Adding the TENSORFLOW_ROOT to the target tool chain path for cortex_m_generic

### PR DESCRIPTION
**Pull request by cdknorow**
- adding toolchain root so that the generated project can be set to the downloaded toolchain.

make -f tensorflow/lite/micro/tools/make/Makefile generate_hello_world_make_project TARGET=cortex_m_generic TARGET_ARCH=cortex-m4+fp  TENSORFLOW_ROOT=/path/to/tflite-micro

the generated makefile will now have the correct path for the compiler

in tools/make/gen/cortexm4_cortex-m4_micro/prj/xxxx/Makefile

```

  TARGET_TOOLCHAIN_ROOT := <path-to-tflite-micro>/tensorflow/lite/micro/tools/make/downloads/gcc_embedded/bin/
  TARGET_TOOLCHAIN_PREFIX := arm-none-eabi-
  
  # These are microcontroller-specific rules for converting the ELF output
  # of the linker into a binary image that can be loaded directly.
  CXX             := '$(TARGET_TOOLCHAIN_ROOT)$(TARGET_TOOLCHAIN_PREFIX)g++'
  CC              := '$(TARGET_TOOLCHAIN_ROOT)$(TARGET_TOOLCHAIN_PREFIX)gcc'
  AS              := '$(TARGET_TOOLCHAIN_ROOT)$(TARGET_TOOLCHAIN_PREFIX)as'
  AR              := '$(TARGET_TOOLCHAIN_ROOT)$(TARGET_TOOLCHAIN_PREFIX)ar'
  LD              := '$(TARGET_TOOLCHAIN_ROOT)$(TARGET_TOOLCHAIN_PREFIX)ld'
  NM              := '$(TARGET_TOOLCHAIN_ROOT)$(TARGET_TOOLCHAIN_PREFIX)nm'
  OBJDUMP         := '$(TARGET_TOOLCHAIN_ROOT)$(TARGET_TOOLCHAIN_PREFIX)objdump'
  OBJCOPY         := '$(TARGET_TOOLCHAIN_ROOT)$(TARGET_TOOLCHAIN_PREFIX)objcopy'
  SIZE            := '$(TARGET_TOOLCHAIN_ROOT)$(TARGET_TOOLCHAIN_PREFIX)size'

```

BUG=fix makefile project generation.

### CODE REVIEW

**Comment by cdknorow** [comment]
@googlebot I consent.

**Comment by advaitjain** [comment]
Once the CLA is signed, we can merge this change.

I would like to point out that we are going to be removing support for makefile based project generation in the near future.

Here is documentation on how to use the Python project generation script instead: https://github.com/tensorflow/tflite-micro/blob/main/tensorflow/lite/micro/docs/new_platform_support.md

And a couple of related issues:
 * https://github.com/tensorflow/tflite-micro/issues/407
 * https://github.com/tensorflow/tflite-micro/issues/445

**Comment by cdknorow** [comment]
@advaitjain thanks for that info on the new build system. I just started the process of migrating our build system to use the tflite-micro repo instead of the tensorflow repo. So far only a couple of minor changes that I needed to make on our side to make things work.

**Comment by cdknorow** [comment]
@advaitjain  Is there anything else I need to do for the CLA. I know I have signed them already for past merges.

**Review by advaitjain — APPROVED** [review]

File: tensorflow/lite/micro/tools/make/targets/cortex_m_generic_makefile.inc
Comment (advaitjain) [inline tensorflow/lite/micro/tools/make/targets/cortex_m_generic_makefile.inc:118]: ah, just noticed this.

This should be:
```
${TENSORFLOW_ROOT}$(MAKEFILE_DIR)/...
```

i.e. without the / after `TENSORFLOW_ROOT` so that when `TENSORFLOW_ROOT` is empty we still have the correct path.

**Review by advaitjain — CHANGES_REQUESTED** [review]

**Comment by advaitjain** [comment]
> @advaitjain thanks for that info on the new build system. I just started the process of migrating our build system to use the tflite-micro repo instead of the tensorflow repo. So far only a couple of minor changes that I needed to make on our side to make things work.

I think my comment about the CLA needing to be singed had a race condition with your comment with `@googlebot I consent.`. Since the `cla:yes` label is on this PR, we are good to go.

**Comment by advaitjain** [comment]
Also, please note that since your're sending a PR from a Github organization account, TFLM maintainers do not have the ability to update the PR. So there may be a bit of trial and error in order to get your PR to be up to date and first in the merge queue.

See other users facing the same issue:
 * https://github.com/tensorflow/tflite-micro/pull/227#issuecomment-880757254
 * https://github.com/tensorflow/tflite-micro/pull/303#issuecomment-883731060

File: tensorflow/lite/micro/tools/make/targets/cortex_m_generic_makefile.inc
Comment (cdknorow) [reply tensorflow/lite/micro/tools/make/targets/cortex_m_generic_makefile.inc]: good catch!

**Review by advaitjain — DISMISSED** [review]

**Comment by advaitjain** [comment]
Approved. The only bit that remains is for you to merge main into your branch. I can not do this since the PR is coming from a github orgranization account.

**Comment by cdknorow** [comment]
@advaitjain  ok, went ahead and rebased onto main.

**Review by advaitjain — APPROVED** [review]

**Pull request merged**
